pkg/common/jobcontroller/jobcontroller.go

```diff
@@ -197,6 +199,34 @@ func (jc *JobController) GenLabels(jobName string) map[string]string {
 	}
 }
 
+func (jc *JobController) SyncPodGroup(job metav1.Object, minAvailableReplicas int32) (*v1alpha1.PodGroup, error) {
+
+	kubeBatchClientInterface := kbutil.GetKubeBatchClientInterface()
+	// Check whether podGroup exists or not
+	podGroup, err := kubeBatchClientInterface.SchedulingV1alpha1().PodGroups(job.GetNamespace()).Get(job.GetName(), metav1.GetOptions{})
+	if err == nil || !k8serrors.IsNotFound(err) {
+		if err == nil {
```

Comment:
Confusing error check

Reply:
Have used same error check done for SyncPdb method

Reply:
I also find this confusing to read. Why not just split into 2 if-blocks?

Reply:
Done